File: Arrays/Rotate Array/rotate.cpp
class Solution {
  public:

    // Function to rotate an array by d elements in counter-clockwise direction.
    void reverse(vector<int>& arr, int start, int end)
    {
        while(start<end)
        {
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
    }
    
    void rotateArr(vector<int>& arr, int d) {
        // code here
        int n = arr.size();
        
        d = d%n;
        
        reverse(arr,0,d-1);
        reverse(arr,d,n-1);
        reverse(arr,0,n-1);
    }
};

File: Arrays/Rotate Array/rotate.py
#User function Template for python3

class Solution:
    #Function to rotate an array by d elements in counter-clockwise direction. 
    
    def reversearr(self,arr,i,j):
        while(i<j):
            arr[i],arr[j] = arr[j],arr[i]
            i+=1
            j-=1
    
    def rotateArr(self, arr, d):
        #Your code here
        n = len(arr)
        d = d%n
        
        self.reversearr(arr,0,d-1);
        self.reversearr(arr,d,n-1);
        self.reversearr(arr,0,n-1);
        
        
